source/common/upstream/eds.cc

```diff
@@ -121,6 +121,7 @@ bool EdsClusterImpl::updateHostsPerLocality(HostSet& host_set, const HostVector&
 
   HostVector hosts_added;
   HostVector hosts_removed;
+  const bool depend_on_hc = health_checker_ != nullptr && !info()->drainConnectionsOnEdsRemoval();
```

Comment:
I think this influences other aspects of cluster initialization as well (marking the cluster as unhealthy until we get the first HC), which we probably don't want to do. See the `upstream_impl.cc` relevant code.

Reply:
Ah yeah I see that. Would a reasonable approach be to add another boolean flag to `updateDynamicHostList` to the effect of `always_delete` which skips the part checking for `FAILED_ACTIVE_HC` when set to true?

Reply:
I would play around with as to whether adding a new boolean or switching to an enum is cleaner. Also, I would probably consider renaming `depend_on_hc` to something else as I think with a 2nd boolean it would be pretty confusing. In general though, this seems like the right path. Please make sure you add a test that would have caught the init issue. (Great catch @htuch!)

Reply:
Added a second boolean, renamed the existing one and added asserts to the test to verify that endpoints are initialized to `FAILED_ACTIVE_HC`